Module/STIG/Convert/Functions.PowerStigXml.ps1

```diff
@@ -698,4 +707,169 @@ function Get-StigObjectsWithOrgSettings
+#>
+function Get-HardCodedRuleLogFileEntry
+{
+    [CmdletBinding()]
+    [OutputType([string])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $RuleId
+    )
+    DynamicParam {
+        Get-DynamicParameterRuleTypeName
+    }
+
+    begin
+    {
+        # Bind the specified parameter values to RuleType var
+        $RuleType = $PSBoundParameters['RuleType']
+        $counter = 0
+
+        # Dynamically query the base rule common properties to remove
+        $commonPropertiesToRemove = Get-BaseRulePropertyName
+
+        # Log file patterns to build log file string
+        $logFileRuleId = '{0}::*::' -f $RuleId
+        $logFileHardCodedRulePattern = "{0}HardCodedRule({1}){4}DscResource = '{2}'{3}{5}"
+        $keyValuePairPattern = '; {0} = $null'
+        $splitRulePattern = '<splitRule>'
+        $open, $close = '@{', '}'
+    }
+
+    process
+    {
+        $returnString = foreach ($type in $RuleType)
+        {
+            # Create convert rule of the given type in order to obtain rule specific properties
+            $ruleTypeConvert = New-Object -TypeName ("$type`Convert")
+            $ruleTypeConvert.SetDscResource()
+            $ruleTypeDscResource = $ruleTypeConvert.DscResource
+
+            # Query all valid non-base rule property names
+            $ruleProperties = (Get-Member -InputObject $ruleTypeConvert -MemberType Property).Name |
+                Where-Object -FilterScript {$PSItem -notin $commonPropertiesToRemove}
+
+            # Build a string for DSC Resource specific parameters, without values
+            $keyValuePair = @()
+            foreach ($dscKey in $ruleProperties)
+            {
+                $keyValuePair += $keyValuePairPattern -f $dscKey
+            }
+            $keyValuePair = -join $keyValuePair
+
+            # First time through, add the rule id, second and more will add the split delimiter
+            if ($counter -eq 0)
+            {
+                $logFileHardCodedRulePattern -f $logFileRuleId, $type, $ruleTypeDscResource, $keyValuePair, $open, $close
+                $counter++
+            }
+            else
+            {
+                $logFileHardCodedRulePattern -f $splitRulePattern, $type, $ruleTypeDscResource, $keyValuePair, $open, $close
+            }
+        }
+        return -join $returnString
+    }
+}
+
+<#
+    .SYNOPSIS
+        Helper function to return the base rule property names.
+#>
+function Get-BaseRulePropertyName
+{
+    [CmdletBinding()]
+    [OutputType([string[]])]
+    param()
+
+    $baseRule = [Rule]::new()
+    return (Get-Member -InputObject $baseRule -MemberType Property).Name
+}
+
+<#
+    .SYNOPSIS
+        Helper function to query all RuleTypeNames from the Module folder
+        and return a RuntimeDefinedParameterDictionary for dynamic parameter
+        use.
+#>
+function Get-DynamicParameterRuleTypeName
+{
+    [CmdletBinding()]
```

Comment:
Can we specify the outputType here to keep inline with the style guidelines or does that break this?

Reply:
Done